### PR TITLE
4881 Docket Record Index

### PR DESCRIPTION
**Pull request by matthopson**

### CODE REVIEW

File: shared/src/business/entities/DocketRecord.js [hunk 1]
```diff
@@ -115,6 +114,77 @@ DocketRecord.VALIDATION_RULES = joi.object().keys({
     .description('Served parties code to override system-computed code.'),
 });
 
+DocketRecord.VALIDATION_RULES = joi.object().keys({
```
Comment (rachaelparris) [inline shared/src/business/entities/DocketRecord.js]: Is this a duplicate? It looks like it's defined above on line 46 as well.

File: shared/src/business/useCases/editDocketEntry/completeDocketEntryQCInteractor.js [hunk 2]
```diff
@@ -185,6 +185,7 @@ exports.completeDocketEntryQCInteractor = async ({
       filingDate: updatedDocument.receivedAt,
     },
     { applicationContext },
+    true, // should update the index
```
Comment (rachaelparris) [inline shared/src/business/useCases/editDocketEntry/completeDocketEntryQCInteractor.js]: How does this update the index?

File: shared/src/business/useCases/updatePetitionDetailsInteractor.test.js [hunk 3]
```diff
@@ -133,19 +133,11 @@ describe('updatePetitionDetailsInteractor', () => {
...
-      editState: undefined,
-      entityName: 'DocketRecord',
-      eventCode: 'FEEW',
-      filedBy: undefined,
-      filingDate: '2019-11-30T09:10:11.000Z',
-      index: 4,
-      signatory: undefined,
-      status: undefined,
-    });
+    const wavedDocument = result.docketRecord.find(
```
Comment (rachaelparris) [inline shared/src/business/useCases/updatePetitionDetailsInteractor.test.js]: Nit: `waivedDocument`

File: shared/src/business/utilities/shouldGenerateDocketRecordIndex.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  UNSERVABLE_EVENT_CODES,
+} = require('../entities/EntityConstants');
+
+/**
+ * determines if a docket record entry should get an index
+ *
+ * @param {string} applicationContext the application context
+ * @param {string} caseDetail the case detail object
+ * @param {string} docketRecordEntry the docket record entry
+ * @returns {Promise} the template with the brackets replaced with replacement values
```
Comment (rachaelparris) [inline shared/src/business/utilities/shouldGenerateDocketRecordIndex.js]: Nit: jsdoc

File: shared/src/business/entities/DocketRecord.js (hunk 1 above)
Comment (matthopson) [reply shared/src/business/entities/DocketRecord.js]: Yeah, bad merge conflict resolution.

File: shared/src/business/useCases/editDocketEntry/completeDocketEntryQCInteractor.js (hunk 2 above)
Comment (matthopson) [reply shared/src/business/useCases/editDocketEntry/completeDocketEntryQCInteractor.js]: This is leftover from a refactor. Removing.

File: shared/src/business/utilities/shouldGenerateDocketRecordIndex.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+ */
+const shouldGenerateDocketRecordIndex = ({ caseDetail, docketRecordEntry }) => {
```
Comment (rachaelparris) [inline shared/src/business/utilities/shouldGenerateDocketRecordIndex.js:16]: I found this code pretty hard to follow. I won't hold up the PR for it, but I think it could be cleaned up as part of DOD refactoring. I can help with that if you'd like!

**Review by rachaelparris — APPROVED** [review]

File: shared/src/business/utilities/shouldGenerateDocketRecordIndex.js (hunk 5 above)
Comment (matthopson) [reply shared/src/business/utilities/shouldGenerateDocketRecordIndex.js:16]: Yeah, I'll ping you today to look at it with me. Thanks!
